code/modules/recycling/conveyor2.dm

```diff
@@ -138,24 +136,35 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 	use_power(6)
 
 	//get the first 30 items in contents
-	affecting = list()
-	var/i = 0
 	var/list/items = loc.contents - src
-	for(var/item in items)
-		i++ // we're sure it's a real target to move at this point
-		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
-			break
-		affecting.Add(item)
-
+	if(!LAZYLEN(items))//Dont do anything at all if theres nothing there but the conveyor
```

Comment:
Won't this be false 99.9% of the time due to lighting objects.

Reply:
maybe 
`var/list/items = loc.contents - src - lighting_object` 
can help

Reply:
I'll do that